core/sessions/orm.go

```diff
@@ -224,6 +225,17 @@ func (o *orm) SetPassword(user *User, newPassword string) error {
 	return o.db.Get(user, sql, hashedPassword, user.Email)
 }
 
+func (o *orm) GenerateAuthToken(user *User) (*auth.Token, error) {
```

Comment:
Minor: the name of this method suggests that it's pure and not setting anything in the DB (i.e it's just generating a token and not updating a row in the database). Why not call `SetAuthToken(user, auth.NewToken())` instead?

Reply:
Difficult to mock and control which token I would want to actually set in the DB without relying on randomness when generating the new token, hence why I created an ORM method that encapsulates the random part in order to mock and test properly...

Reply:
That's a good point about mocking and testing. Maybe call this `CreateAndSetAuthToken`? Kind of a mouthful but clearer in it's intentions.